--- a/src/include/c_api/helpers.h
+++ b/src/include/c_api/helpers.h
@@ -1,3 +1,4 @@
+#include <cstdint>
 #pragma once

 #include "c_api/helpers/types.h"
--- a/src/c_api/helpers.cpp
+++ b/src/c_api/helpers.cpp
@@ -1,3 +1,4 @@
+#include <cstdint>
 #include "c_api/helpers.h"

 #include "common/types/value.h"
--- a/src/common/file_system/local_file_system.cpp
+++ b/src/common/file_system/local_file_system.cpp
@@ -1,3 +1,4 @@
+#include <sys/stat.h>
 #include "common/file_system/local_file_system.h"

 #include <fcntl.h>
--- a/src/storage/buffer_manager/buffer_manager.cpp
+++ b/src/storage/buffer_manager/buffer_manager.cpp
@@ -19,7 +19,7 @@
 #include <span>

-#if defined(_WIN32)
+#if defined(_WIN32) && defined(_MSC_VER)
 #include <exception>

 #include <eh.h>
